bin/echo.c

```diff
@@ -165,7 +165,9 @@ int echo(struct s2n_connection *conn, int sockfd)
                 s2n_errno = S2N_ERR_T_OK;
                 bytes_read = s2n_recv(conn, buffer, STDIO_BUFSIZE, &blocked);
                 if (bytes_read == 0) {
-                    return 0;
+                    if (blocked == S2N_NOT_BLOCKED) {
```

Comment:
Should this just be if (bytes_read == 0 && blocked == S2N_NOT_BLOCKED)? I don't think it makes it harder to understand, and it reduces the nested ifs.

And do we even need to consider the bytes_read? Wouldn't looking at the blocked status be enough to determine whether or not we're done?

Reply:
Going through this a little more I think we don't need to consider S2N_NOT_BLOCKED here. The only time that gets set is when the connection is closed. If we return any bytes, we don't enter this conditional anyway. Maybe just leave this as I found it.